### DISCUSSION TITLE
Delete a record without prior loading in Entity Framework

### DISCUSSION THREAD

**Discussion by giscus** [linkdotnet/Blog.Discussions#32] [General]
# Delete a record without prior loading in Entity Framework

Sometimes you have an Id of an object and want to delete the underlying thing from the database. But it doesn't make sense to load the whole object from the database to memory first. So how can we achieve this quickly?

https://steven-giesel.com/blogPost/5f58e2a2-2b50-4e0c-8062-09d003c729a9

**Comment by latonz** [comment]
With entity framework core 7 and its new `ExecuteDelete` feature, this is now as simple as `await dbContext.Objects.Where(x => x.Id == id).ExecuteDeleteAsync();`. See also [What’s new in ef core 7](https://learn.microsoft.com/en-us/ef/core/what-is-new/ef-core-7.0/whatsnew#executeupdate-and-executedelete-bulk-updates).

**Reply by linkdotnet** [reply]
Very good point. I will add this as well (including crediting you)
